servers/republik/modules/crowdfundings/lib/AutoPay.js

```diff
@@ -0,0 +1,220 @@
+const debug = require('debug')('crowdfundings:lib:AutoPay')
+
+const { applyPgInterval: { add: addInterval } } = require('@orbiting/backend-modules-utils')
+
+const createCharge = require('./payments/stripe/createCharge')
+const { getCustomPackages } = require('./User')
+const { getLastEndDate } = require('./utils')
+const createCache = require('./cache')
+
+const suggest = async (membershipId, pgdb) => {
+  // Find membership
+  const membership = await pgdb.public.memberships.findOne({ id: membershipId })
+
+  if (!membership) {
+    return false
+  }
+
+  // Find current periods
+  const membershipPeriods = await pgdb.public.membershipPeriods.find({ membershipId: membership.id })
+
+  // Find pledgeOptions
+  const relatedPledgeOptions = await pgdb.public.pledgeOptions.find(
+    {
+      or: [
+        { membershipId },
+        { pledgeId: membership.pledgeId }
+      ]
+    }
+  )
+
+  if (relatedPledgeOptions.length < 1) {
+    return false
+  }
+
+  // Find latest pledge
+  const pledge = await pgdb.public.pledges.findOne(
+    { id: relatedPledgeOptions.map(po => po.pledgeId), status: 'SUCCESSFUL' },
+    { orderBy: { createdAt: 'DESC' }, limit: 1 }
+  )
+
```

Comment:
handle if no pledge is found here? because it could find a pledge options from a not successful pledge (e.g. draft pledge for postfinance or paypal)